webapp/security/views.py

```diff
@@ -354,6 +359,22 @@ def cve_index():
         .all()
     )
 
+    cve_tree = defaultdict(dict)
+    for cve in cves:
+        packages = cve.status_tree
+
+        if package:
+            packages = {k: v for k, v in packages.items() if k == package}
+
+        if component:
+            packages = {
+                k: v
+                for k, v in packages.items()
+                if any(component == a.component for a in v)
+            }
+
+        cve_tree[cve.id] = packages
```

Comment:
Ideally we should do this in the original `cves_query`, it could be expensive to loop through thousands of packages, @jpmartinspt any ideas?